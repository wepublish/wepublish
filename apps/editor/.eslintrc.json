{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "plugins": ["i18next", "simple-import-sort"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "no-empty": "warn",
        "react/display-name": [0],
        "react/jsx-boolean-value": [1],
        "react/prop-types": [0],
        "i18next/no-literal-string": [
          2,
          {
            "markupOnly": true,
            "ignoreAttribute": [
              "align",
              "alignItems",
              "appearance",
              "as",
              "autoComplete",
              "axis",
              "backdrop",
              "closePath",
              "color",
              "componentClass",
              "controlId",
              "data-testid",
              "dataKey",
              "dateFormat",
              "display",
              "element",
              "eventKey",
              "fill",
              "fixed",
              "fixed",
              "flexDirection",
              "flexWrap",
              "format",
              "href",
              "icon",
              "iconActive",
              "justify",
              "justifyContent",
              "minHeight",
              "name",
              "overflow",
              "path",
              "placement",
              "position",
              "sdkLanguage",
              "size",
              "spacing",
              "stack",
              "target",
              "to",
              "trigger",
              "value",
              "variant",
              "verticalAlign"
            ],
            "ignoreComponent": ["code"],
            "ignore": ["â€”", "&"]
          }
        ],
        "simple-import-sort/imports": 1,
        "simple-import-sort/exports": 1,
        "import/first": 1,
        "import/newline-after-import": 1,
        "import/no-duplicates": 1
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
