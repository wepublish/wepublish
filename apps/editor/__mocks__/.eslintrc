{
  "rules": {
    "@typescript-eslint/ban-ts-comment": "off",
    "@typescript-eslint/no-var-requires": "off"
  }
}
