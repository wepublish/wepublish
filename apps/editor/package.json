{
  "name": "@wepublish/editor",
  "version": "9.2.0",
  "description": "We.publish CMS Editor",
  "keywords": [
    "editor",
    "react",
    "postgres",
    "publishing",
    "media",
    "newspaper",
    "cms",
    "subscription"
  ],
  "homepage": "https://github.com/wepublish/wepublish",
  "bugs": {
    "url": "https://github.com/wepublish/wepublish/issues",
    "email": "development@wepublish.ch"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/wepublish/wepublish.git",
    "directory": "apps/editor"
  },
  "bin": "main.js",
  "main": "main.js",
  "license": "MIT"
}
