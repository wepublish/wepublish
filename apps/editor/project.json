{
  "name": "editor",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "apps/editor/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "nx:run-commands",
      "options": {
        "commands": [
          {
            "command": "npx nx server editor"
          },
          {
            "command": "npx nx browser editor"
          }
        ],
        "parallel": false
      }
    },
    "serve-ssr": {
      "executor": "@nx/webpack:ssr-dev-server",
      "defaultConfiguration": "development",
      "options": {
        "browserTarget": "editor:build:development",
        "serverTarget": "editor:serve-server:development",
        "port": 3000,
        "browserTargetOptions": {
          "watch": true
        }
      },
      "configurations": {
        "development": {},
        "production": {
          "browserTarget": "editor:build:production",
          "serverTarget": "editor:serve-server:production"
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["apps/editor/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "apps/editor/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "browser": {
      "executor": "@nx/webpack:webpack",
      "outputs": ["{options.outputPath}"],
      "defaultConfiguration": "production",
      "options": {
        "compiler": "babel",
        "outputPath": "dist/apps/editor/browser",
        "index": "apps/editor/src/index.html",
        "baseHref": "/",
        "main": "apps/editor/src/main.tsx",
        "polyfills": "apps/editor/src/polyfills.ts",
        "tsConfig": "apps/editor/tsconfig.app.json",
        "assets": ["apps/editor/src/assets"],
        "styles": [],
        "scripts": [],
        "webpackConfig": "apps/editor/webpack.config.js",
        "isolatedConfig": true
      },
      "configurations": {
        "development": {
          "extractLicenses": false,
          "optimization": false,
          "sourceMap": true,
          "vendorChunk": true
        },
        "production": {
          "fileReplacements": [
            {
              "replace": "apps/editor/src/environments/environment.ts",
              "with": "apps/editor/src/environments/environment.prod.ts"
            }
          ],
          "optimization": true,
          "outputHashing": "all",
          "sourceMap": false,
          "namedChunks": false,
          "extractLicenses": true,
          "vendorChunk": false
        }
      }
    },
    "server": {
      "executor": "@nx/webpack:webpack",
      "outputs": ["{options.outputPath}"],
      "defaultConfiguration": "production",
      "options": {
        "target": "node",
        "main": "apps/editor/server.ts",
        "outputPath": "dist/apps/editor",
        "tsConfig": "apps/editor/tsconfig.server.json",
        "compiler": "babel",
        "externalDependencies": "all",
        "generatePackageJson": true,
        "outputHashing": "none",
        "webpackConfig": "apps/editor/webpack.config.server.js",
        "isolatedConfig": true
      },
      "configurations": {
        "development": {
          "optimization": false,
          "sourceMap": true
        },
        "production": {
          "fileReplacements": [
            {
              "replace": "apps/editor/src/environments/environment.ts",
              "with": "apps/editor/src/environments/environment.prod.ts"
            }
          ],
          "sourceMap": false
        }
      }
    },
    "serve": {
      "executor": "@nx/webpack:dev-server",
      "defaultConfiguration": "development",
      "options": {
        "buildTarget": "editor:browser",
        "port": 3000,
        "hmr": true
      },
      "configurations": {
        "development": {
          "buildTarget": "editor:browser:development"
        },
        "production": {
          "buildTarget": "editor:browser:production",
          "hmr": false
        }
      }
    },
    "serve-server": {
      "executor": "@nx/js:node",
      "defaultConfiguration": "development",
      "options": {
        "buildTarget": "editor:server:development",
        "buildTargetOptions": {
          "watch": true
        }
      },
      "configurations": {
        "development": {},
        "production": {
          "buildTarget": "editor:server:production"
        }
      }
    }
  },
  "tags": []
}
