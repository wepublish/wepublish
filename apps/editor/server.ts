import cors from 'cors'
import express from 'express'
import * as path from 'path'

import {handleRequest} from './src/main.server'

if (!process.env.API_URL) throw new Error('No API_URL specified in environment.')

const port = process.env['PORT'] || 3000
const app = express()

const browserDist = path.join(process.cwd(), 'dist/apps/editor/browser')
const indexPath = path.join(browserDist, 'index.html')

app.use(cors())

app.get('*.*', express.static(browserDist, {}))

app.use('*', handleRequest(indexPath))

const server = app.listen(port, () => {
  // Server has started
})

server.on('error', console.error)
