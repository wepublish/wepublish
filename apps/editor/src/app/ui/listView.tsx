import styled from '@emotion/styled'
import {Table} from 'rsuite'

import {ListViewFilters} from '../atoms/searchAndFilter/listViewFilters'
import {StateColor} from '../utility'

const ListViewContainer = styled.div`
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 2fr 1fr;
  justify-content: start;
  align-items: center;
`

const ListViewHeader = styled.div`
  grid-column: 1 2;
`

const ListViewActions = styled.div`
  grid-column: 2;
  text-align: right;
`

const ListFilters = styled(ListViewFilters)`
  grid-column: 1/3;
`

const TableWrapper = styled.div`
  display: 'flex';
  flex-flow: 'column';
  margin-top: '20px';
`

interface StatusBadgeProps {
  states: string[]
}

const StatusBadge = styled.div<StatusBadgeProps>`
  text-align: center;
  border-radius: 15px;
  background-color: ${props => {
    if (props.states.includes('pending')) {
      return StateColor.pending
    } else if (props.states.includes('published')) {
      return StateColor.published
    } else if (props.states.includes('draft')) {
      return StateColor.draft
    } else {
      return StateColor.none
    }
  }};
`

const IconButtonCell = styled(Table.Cell)`
  padding: 6px 0;
  & > div {
    padding: 0;
    button:nth-of-type(n + 1) {
      margin-left: 5px;
    }
  }
`

export {
  IconButtonCell,
  ListFilters,
  ListViewActions,
  ListViewContainer,
  ListViewHeader,
  StatusBadge,
  StatusBadgeProps,
  TableWrapper
}
