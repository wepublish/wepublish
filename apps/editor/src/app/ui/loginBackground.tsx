import styled from '@emotion/styled'

import {Logo} from '../logo'

const BgWrapper = styled.div`
  position: relative;
  width: 340px;
  height: 40px;
  transform: translateY(-0px);
`

const Shape1 = styled.div`
  position: absolute;
  left: 50%;
  width: 260px;
  height: 260px;
  border-radius: 100%;
  transform: translateX(-50%) translateX(-180px) translateY(-40px);
  background: linear-gradient(230deg, #f08c1f 0%, #ffa463 100%);
`

const Shape2 = styled.div`
  position: absolute;
  left: 50%;
  width: 260px;
  height: 260px;
  border-radius: 100%;
  transform: translateX(-50%) translateX(180px) translateY(-40px);
  background: linear-gradient(10deg, #29805a 0%, #34d690 100%);
`

const Shape3 = styled.div`
  position: absolute;
  left: 50%;
  width: 260px;
  height: 260px;
  border-radius: 100%;
  transform: translateX(-50%) translateY(-140px);
  background: linear-gradient(-40deg, #03738c 0%, #04c4d9 100%);
`

const Shape4 = styled.div`
  position: absolute;
  width: 340px;
  height: 340px;
  border-radius: 100%;
  transform: translateY(-80px);
  background: linear-gradient(-90deg, #d95560 0%, #ff6370 100%);
`

const LogoWrapper = styled.div`
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  width: 230px;
`

export const Background = () => (
  <BgWrapper>
    <Shape1 />
    <Shape2 />
    <Shape3 />
    <Shape4 />
    <LogoWrapper>
      <Logo />
    </LogoWrapper>
  </BgWrapper>
)
