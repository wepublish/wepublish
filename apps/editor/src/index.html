<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Editor</title>
    <base href="/" />

    <link
      href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700,700i&display=swap"
      rel="stylesheet"
    />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="icon" type="image/png" sizes="128x128" href="/assets/favicon-128.png" />
    <link rel="icon" type="image/png" sizes="64x64" href="/assets/favicon-64.png" />
  </head>
  <body>
    <noscript>Please enable JavaScript!</noscript>
    <div id="react-root"></div>
  </body>
</html>
