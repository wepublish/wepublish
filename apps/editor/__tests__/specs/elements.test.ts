import {emptyCellsTable} from '@wepublish/ui/editor'

describe('emptyCellsTable', () => {
  test.each([emptyCellsTable(1, 1), emptyCellsTable(2, 2)])(
    'should match snapshot %p',
    createTable => {
      expect(createTable).toMatchSnapshot()
    }
  )
})
