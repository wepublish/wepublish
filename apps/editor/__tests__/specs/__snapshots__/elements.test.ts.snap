// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emptyCellsTable should match snapshot {"children": [Array], "type": "table"} 1`] = `
{
  "children": [
    {
      "children": [
        {
          "borderColor": "#000000",
          "children": [
            {
              "children": [
                {
                  "text": "",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "table-cell",
        },
      ],
      "type": "table-row",
    },
  ],
  "type": "table",
}
`;

exports[`emptyCellsTable should match snapshot {"children": [Array], "type": "table"} 2`] = `
{
  "children": [
    {
      "children": [
        {
          "borderColor": "#000000",
          "children": [
            {
              "children": [
                {
                  "text": "",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "table-cell",
        },
        {
          "borderColor": "#000000",
          "children": [
            {
              "children": [
                {
                  "text": "",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "table-cell",
        },
      ],
      "type": "table-row",
    },
    {
      "children": [
        {
          "borderColor": "#000000",
          "children": [
            {
              "children": [
                {
                  "text": "",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "table-cell",
        },
        {
          "borderColor": "#000000",
          "children": [
            {
              "children": [
                {
                  "text": "",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "table-cell",
        },
      ],
      "type": "table-row",
    },
  ],
  "type": "table",
}
`;
